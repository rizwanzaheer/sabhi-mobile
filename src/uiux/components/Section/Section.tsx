import * as React from 'react'
import { withTheme } from '../../theming'

import Container from '../Container/Container'
import Text, { TextTypes } from '../Text/Text'

interface SectionProps {
  /**
   * Section title
   */
  title?: string

  /**
   * Section title type. Use TextTypes constant
   */
  sectionTitleType?: string

  /**
   * Remove top margin
   */
  noTopMargin?: boolean

  /**
   * Remove top border
   */
  noTopBorder?: boolean

  /**
   * Theme passed in by HOC
   */
  theme: any
}

const Section: React.FunctionComponent<SectionProps> = props => {
  return (
    <Container marginTop={props.noTopMargin ? 0 : props.theme.spacing.section}>
      {props.title && (
        <Container paddingLeft={true}>
          <Text
            paddingBottom={true}
            bold={!!props.sectionTitleType}
            type={props.sectionTitleType ? props.sectionTitleType : TextTypes.SectionHeader}
          >
            {props.title}
          </Text>
        </Container>
      )}
      <Container dividerTop={!props.noTopBorder} dividerBottom={true} background={'primary'}>
        {props.children}
      </Container>
    </Container>
  )
}

export default withTheme(Section)
