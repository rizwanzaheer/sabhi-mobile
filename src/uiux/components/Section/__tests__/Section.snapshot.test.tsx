import * as React from 'react'
import { render } from 'react-native-testing-library'
import Section from '../Section'

describe('Component(snapshot): Section', () => {
  it('should render with no props', () => {
    const tree = render(<Section />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render with just a title', () => {
    const tree = render(<Section title={'Section title'} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render with props', () => {
    const tree = render(
      <Section
        title={'Section title'}
        noTopBorder={true}
        noTopMargin={true}
        sectionTitleType={'sectionHeader'}
      />,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
