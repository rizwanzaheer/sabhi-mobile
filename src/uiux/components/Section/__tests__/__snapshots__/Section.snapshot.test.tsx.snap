// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component(snapshot): Section should render with just a title 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "alignItems": undefined,
      "backgroundColor": undefined,
      "borderRadius": undefined,
      "flex": undefined,
      "flexDirection": undefined,
      "height": undefined,
      "justifyContent": undefined,
      "margin": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": 20,
      "padding": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "width": undefined,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "backgroundColor": undefined,
        "borderRadius": undefined,
        "flex": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": 15,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": undefined,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#BBBBBB",
          "flexWrap": "wrap",
          "fontSize": 14,
          "paddingBottom": 15,
        }
      }
    >
      Section title
    </Text>
  </View>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "backgroundColor": "#FFFFFF",
        "borderBottomColor": "#CCCCCC",
        "borderBottomWidth": 0.5,
        "borderRadius": undefined,
        "borderTopColor": "#CCCCCC",
        "borderTopWidth": 0.5,
        "flex": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": undefined,
      }
    }
  />
</View>
`;

exports[`Component(snapshot): Section should render with no props 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "alignItems": undefined,
      "backgroundColor": undefined,
      "borderRadius": undefined,
      "flex": undefined,
      "flexDirection": undefined,
      "height": undefined,
      "justifyContent": undefined,
      "margin": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": 20,
      "padding": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "width": undefined,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "backgroundColor": "#FFFFFF",
        "borderBottomColor": "#CCCCCC",
        "borderBottomWidth": 0.5,
        "borderRadius": undefined,
        "borderTopColor": "#CCCCCC",
        "borderTopWidth": 0.5,
        "flex": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": undefined,
      }
    }
  />
</View>
`;

exports[`Component(snapshot): Section should render with props 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "alignItems": undefined,
      "backgroundColor": undefined,
      "borderRadius": undefined,
      "flex": undefined,
      "flexDirection": undefined,
      "height": undefined,
      "justifyContent": undefined,
      "margin": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": 0,
      "padding": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "width": undefined,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "backgroundColor": undefined,
        "borderRadius": undefined,
        "flex": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": 15,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": undefined,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#BBBBBB",
          "flexWrap": "wrap",
          "fontSize": 14,
          "fontWeight": "bold",
          "paddingBottom": 15,
        }
      }
    >
      Section title
    </Text>
  </View>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "backgroundColor": "#FFFFFF",
        "borderBottomColor": "#CCCCCC",
        "borderBottomWidth": 0.5,
        "borderRadius": undefined,
        "flex": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": undefined,
      }
    }
  />
</View>
`;
